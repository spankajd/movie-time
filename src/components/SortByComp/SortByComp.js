import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { setSortBy } from "../../store/actions/movieActions";

import './SortByComp.scss';

const SortByComp = () => {

    const dispatch = useDispatch();

    const [sortOption, setSortOption] = useState("");
  
    const handleSortChange = (e) => {
        const selectedOption = e.target.value;
        setSortOption(selectedOption);
        dispatch(setSortBy(selectedOption));
    }

    return <div className="SortByComp">
        <label htmlFor="sortBy">Sort By: </label>
        <select id="sortBy" value={sortOption} onChange={handleSortChange} data-testid="sortBy">
            <option value="">Select</option>
            <option value="episode_id">Episode</option>
            <option value="release_date">Release Date</option>
            <option value="title">Title</option>
            <option value="imdbRating">Rating</option>
        </select>
    </div>
}

export default SortByComp;