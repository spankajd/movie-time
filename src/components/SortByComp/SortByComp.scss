.SortByComp {
    label {
        margin-right: 5px;
        font-weight: 500;
    }

    #sortBy {
        height: 40px;
        padding: 5px;
        font-size: 16px;
        outline: none;
        border: none;
    }   

  @media (max-width: 768px) {
    font-size: 14px;
    #sortBy {
        font-size: 14px;
        height: 32px;
    }
  }
}



