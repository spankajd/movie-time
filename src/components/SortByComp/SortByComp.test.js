import { fireEvent, screen } from "@testing-library/react";
import '@testing-library/jest-dom'; // Provides additional matchers
import { renderWithProviders } from "../../utils/test-utlis";
import SortByComp from "./SortByComp";

describe("SortByComp Component", () => {
    it("renders SortByComp", () => {
        renderWithProviders(
            <SortByComp />
        );
        const element = screen.getByText(/Sort By/i);
        expect(element).toBeInTheDocument();
    });

    it("matches snapshot", () => {
        const { asFragment } = renderWithProviders(<SortByComp />);
        expect(asFragment()).toMatchSnapshot();
    });
    
    it("renders SortByComp with handleSortChange triggered", () => {
        renderWithProviders(<SortByComp />);
        let selecteElement = screen.getByTestId('sortBy');
        // fireEvent.click(selecteElement);
        // const element = screen.getByText("Episode");
        fireEvent.change(selecteElement, {
            target: {
                value:"Episode"
            }
        });
        const element = screen.getByText("Episode");
        expect(element).toBeInTheDocument();

        // expect(selecteElement.value).toBe('Episode');
    });
});