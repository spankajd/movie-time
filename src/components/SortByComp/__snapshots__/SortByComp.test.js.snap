// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortByComp Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SortByComp"
  >
    <label
      for="sortBy"
    >
      Sort By: 
    </label>
    <select
      data-testid="sortBy"
      id="sortBy"
    >
      <option
        value=""
      >
        Select
      </option>
      <option
        value="episode_id"
      >
        Episode
      </option>
      <option
        value="release_date"
      >
        Release Date
      </option>
      <option
        value="title"
      >
        Title
      </option>
      <option
        value="imdbRating"
      >
        Rating
      </option>
    </select>
  </div>
</DocumentFragment>
`;
